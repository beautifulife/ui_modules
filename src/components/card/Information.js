import React from 'react';
import PropTypes from 'prop-types';
import './Information.scss';

const Information = ({ crossOut, hilight, label, title }) => {
  return (
    <div className="Information">
      <p className="Information__label">{label || 'Card Label'}</p>
      <p className="Information__title">{title || 'Card Title'}</p>
      <div className="Information__bottom">
        <span className="hilight">{hilight || 'Hilight'}</span>
        <span className="cross_out">{crossOut || 'Cross Out'}</span>
      </div>
    </div>
  );
};

Information.propTypes = {
  crossOut: PropTypes.string,
  hilight: PropTypes.string,
  label: PropTypes.string,
  title: PropTypes.string
};

Information.defaultProps = {
  crossOut: '',
  hilight: '',
  label: '',
  title: ''
};

export default Information;
