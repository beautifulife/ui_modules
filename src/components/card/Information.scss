.Information {
  padding: 10px;

  &__label {
    margin-bottom: 5px;
    font-size: 0.8rem;
    color: rgba(0, 0, 0, 0.45);
  }

  &__title {
    height: 2.5rem;
  }

  &__bottom {
    .hilight {
      color: red;
      margin-right: 3px;
    }
    .cross_out {
      font-size: 0.65rem;
      color: rgba(0, 0, 0, 0.45);
      text-decoration: line-through;
    }
  }
}
