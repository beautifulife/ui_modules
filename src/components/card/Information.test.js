import React from 'react';
import { shallow } from 'enzyme';
import Information from './Information';

describe('Information', () => {
  describe('about render', () => {
    const wrapper = shallow(<Information />);

    it('compare with snapshot', () => {
      expect(wrapper).toMatchSnapshot();
    });
  });
});
