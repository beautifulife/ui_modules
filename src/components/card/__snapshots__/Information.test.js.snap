// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Information about render compare with snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Information
    crossOut=""
    hilight=""
    label=""
    title=""
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <p
          className="Information__label"
        >
          Card Label
        </p>,
        <p
          className="Information__title"
        >
          Card Title
        </p>,
        <div
          className="Information__bottom"
        >
          <span
            className="hilight"
          >
            Hilight
          </span>
          <span
            className="cross_out"
          >
            Cross Out
          </span>
        </div>,
      ],
      "className": "Information",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Card Label",
          "className": "Information__label",
        },
        "ref": null,
        "rendered": "Card Label",
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Card Title",
          "className": "Information__title",
        },
        "ref": null,
        "rendered": "Card Title",
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <span
              className="hilight"
            >
              Hilight
            </span>,
            <span
              className="cross_out"
            >
              Cross Out
            </span>,
          ],
          "className": "Information__bottom",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Hilight",
              "className": "hilight",
            },
            "ref": null,
            "rendered": "Hilight",
            "type": "span",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Cross Out",
              "className": "cross_out",
            },
            "ref": null,
            "rendered": "Cross Out",
            "type": "span",
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <p
            className="Information__label"
          >
            Card Label
          </p>,
          <p
            className="Information__title"
          >
            Card Title
          </p>,
          <div
            className="Information__bottom"
          >
            <span
              className="hilight"
            >
              Hilight
            </span>
            <span
              className="cross_out"
            >
              Cross Out
            </span>
          </div>,
        ],
        "className": "Information",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Card Label",
            "className": "Information__label",
          },
          "ref": null,
          "rendered": "Card Label",
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Card Title",
            "className": "Information__title",
          },
          "ref": null,
          "rendered": "Card Title",
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <span
                className="hilight"
              >
                Hilight
              </span>,
              <span
                className="cross_out"
              >
                Cross Out
              </span>,
            ],
            "className": "Information__bottom",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Hilight",
                "className": "hilight",
              },
              "ref": null,
              "rendered": "Hilight",
              "type": "span",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Cross Out",
                "className": "cross_out",
              },
              "ref": null,
              "rendered": "Cross Out",
              "type": "span",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
